src/pydicom/pixels/utils.py

```diff
@@ -1276,7 +1276,7 @@ def _passes_version_check(package_name: str, minimum_version: tuple[int, ...]) -
         return tuple(int(x) for x in module.__version__.split(".")) >= minimum_version
     except Exception as exc:
         if config.debugging:
```

Comment:
Still the same question: why do you need to check for `config.debugging` here (or elsewhere)?

Reply:
It's probably not as useful here, but in the `data_element_generator` code especially it's used to prevent logging in heavily used parts of the code base, presumably for efficiency.

Reply:
Removed

Reply:
Thanks, that makes some sense, though I would still argue that this could be just debug logging.

Reply:
> It's probably not as useful here, but in the `data_element_generator` code especially it's used to prevent logging in heavily used parts of the code base, presumably for efficiency.

This. Could probably do it by checking the debug level as well, but actually calling `LOGGING.debug()` would be an "expensive"  function call regardless of logging level. An `if` statement is much faster.

Reply:
Ah, I see. Actually, checking the debug level ist the first thing that the logger does, but it is of course more expansive than the `if` statement. Also, we would have to make sure, that we don't do anything expansive in the log message that will evaluated during the call.
So this is actually an optimization that can be used in critical code paths.